Search result path not correct

### Please describe the bug you are seeing

Hi,

I installed this plugin to an empty docusaurus project and run all the command without any customization or modifications. When i build and serve the project the search works and shows results fine, but the path where the result is pointing is not working correct.

I dont know if this is a undocumentened config or bug.

![Screenshot 2024-07-05 113950](https://github.com/cmfcmf/docusaurus-search-local/assets/123746320/3c7c4b2f-636f-4f7c-8e7a-123b1c462c32)

As you can see from the image the path is duplicating after the docs part.

"https://symmetrical-space-doodle-x5rrrgx9pq4hv74v-3000.app.github.dev/docs/tutorial-extras/translate-your-site/docs/tutorial-extras/translate-your-site#start-your-localized-site"

When it should be:

"https://symmetrical-space-doodle-x5rrrgx9pq4hv74v-3000.app.github.dev/docs/tutorial-extras/translate-your-site#start-your-localized-site"

Thanks!

### How can we best reproduce the bug?

1. Install docusaurus
2. Install plugin with default settings
3. Test

### Which version of Docusaurus and @cmfcmf/docusaurus-search-local are you using?

Latest  that are available

### Which Node.js version are you using?

22.3.0

### Which theme are you using?

Yes, I am using @docusaurus/theme-classic

### How does your plugin config look like?

Default config

### Additional context

_No response_

Docusaurus v3.5.2 has been released and should work fine with this plugin (I tried with v1.2)

---

Note: you may get another error while building your site.

```text
[ERROR] Error: Unable to build website for locale en.
  ...
  [cause]: TypeError: Cannot read properties of undefined (reading 'load')
      at getDocusaurusTag (.../@cmfcmf/docusaurus-search-local/lib/server/parse.js:194:33)
      at .../@cmfcmf/docusaurus-search-local/lib/server/index.js:288:60
```

It is a different issue, related to a recent release of Cheerio v1.0 that did unexpected breaking changes. See my PR for details and workarounds: https://github.com/cmfcmf/docusaurus-search-local/pull/218

--- a/packages/docusaurus-search-local/src/server/parse.test.js
+++ b/packages/docusaurus-search-local/src/server/parse.test.js
@@ -310,5 +310,43 @@ describe("parser", () => {
         ],
       });
     });
+
+    it("parses hashes correctly from URLs", () => {
+      const html = `<html>
+        <head><title>TITLE</title></head>
+        <body>
+          <article>
+            <h1>a<a href="lala.com#first#second" class="hash-link"></a></h1>
+            <h1>b<a href="#first#second" class="hash-link"></a></h1>
+            <h1>c<a href="/foo#first#second" class="hash-link"></a></h1>
+          </article>
+        </body>
+        </html>
+      `;
+      expect(html2text(html, "docs")).toEqual({
+        docSidebarParentCategories: [],
+        pageTitle: "a",
+        sections: [
+          {
+            title: "a",
+            hash: "#first#second",
+            content: "",
+            tags: [],
+          },
+          {
+            title: "b",
+            hash: "#first#second",
+            content: "",
+            tags: [],
+          },
+          {
+            title: "c",
+            hash: "#first#second",
+            content: "",
+            tags: [],
+          },
+        ],
+      });
+    });
   });
 });

--- a/packages/docusaurus-search-local/src/server/parse.ts
+++ b/packages/docusaurus-search-local/src/server/parse.ts
@@ -124,8 +124,9 @@ export function html2text(
           // <a class="hash-link" href="#first-header" title="Direct link to heading">#</a>
           .not("a[aria-hidden=true], a.hash-link")
           .text();
-        const hash = $(heading).find("a.hash-link").attr("href") || "";
-
+        const linkHash = $(heading).find("a.hash-link").attr("href") || "";
+        const [, hashPart] = linkHash.split("#");
+        const hash = hashPart ? `#${hashPart}` : "";
         let $sectionElements;
         if ($(heading).parents(".markdown").length === 0) {
           // $(heading) is the page title

--- a/packages/docusaurus-search-local/src/server/parse.ts
+++ b/packages/docusaurus-search-local/src/server/parse.ts
@@ -125,8 +125,8 @@ export function html2text(
           .not("a[aria-hidden=true], a.hash-link")
           .text();
         const linkHash = $(heading).find("a.hash-link").attr("href") || "";
-        const [, hashPart] = linkHash.split("#");
-        const hash = hashPart ? `#${hashPart}` : "";
+        const [, ...hashParts] = linkHash.split("#");
+        const hash = hashParts.length ? `#${hashParts.join("#")}` : "";
         let $sectionElements;
         if ($(heading).parents(".markdown").length === 0) {
           // $(heading) is the page title

